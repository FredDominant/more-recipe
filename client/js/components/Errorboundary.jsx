import React from 'react';
import PropTypes from 'prop-types';

import Footer from './Footer';

/**
 *
 * @class ErrorBoundary
 *
 * @extends {Component}
 */
class Errorboundary extends React.Component {
/**
 * @description Creates an instance of ErrorBoundary.
 *
 * @memberof ErrorBoundary
 *
 * @param {object} props
 */
  constructor(props) {
    super(props);
    this.state = {
      hasError: false
    };
  }
  /**
   * @memberof Errorboundary
   *
   * @returns {null} null
   *
   */
  componentDidCatch() {
    this.setState({
      hasError: true
    });
  }
  /**
   *
   * @memberof Errorboundary
   *
  * @return {ReactElement} markup
   */
  render() {
    if (this.state.hasError) {
      return (
        <div>
          <br />
          <div className="container">
            <h3 className="text-center allRecipes-title">
              An Error has ocurred. Please try again later
            </h3>
            <br />
          </div>
          <br />
          <Footer />
        </div>
      );
    }
    return this.props.children;
  }
}
Errorboundary.propTypes = {
  children: PropTypes.node.isRequired
};

export default Errorboundary;
