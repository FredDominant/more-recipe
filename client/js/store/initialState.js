import authenticateUser from '../utils/authenticateUser';

const initialState = {
  auth: {
    isAuthenticated: authenticateUser()
  },
  isFetching: false,
  isUploading: false,
  recipes: {
  }
};
export default initialState;
