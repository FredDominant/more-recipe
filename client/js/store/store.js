const store = {
  loginForm: {
    email: '',
    password: ''
  },
  signunForm: {
    firstname: '',
    lastname: '',
    email: '',
    password: '',
    confirmPassword: ''
  }
};
export default store;
