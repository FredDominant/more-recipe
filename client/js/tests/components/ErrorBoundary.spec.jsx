import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import store from '../../store/store';

import Errorboundary from '../../components/Errorboundary';

const props = {
  children: <div />
};
/**
 * @description test for Errorboundary component
 *
 * @returns {null} null
 */
function setup() {
  const shallowWrapper = shallow(<Errorboundary {...props} />);
  const mountedWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <Errorboundary {...props} store={store} />
      </BrowserRouter>
    </Provider>
  );

  return { shallowWrapper, mountedWrapper };
}
describe('ErrorBoundary component should', () => {
  it('render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('render correctly if error', () => {
    const wrapper = shallow(<Errorboundary {...props} />);
    wrapper.setState({ hasError: true });
    expect(wrapper.find('div').length).toBe(2);
  });

  it('call componentDidCatch() if error occurs', () => {
    const wrapper = shallow(<Errorboundary {...props} />);
    wrapper.setState({ hasError: true });
    wrapper.instance().componentDidCatch();
    expect(wrapper).toMatchSnapshot();
  });
});
