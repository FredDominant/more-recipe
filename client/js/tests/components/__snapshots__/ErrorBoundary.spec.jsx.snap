// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component should call componentDidCatch() if error occurs 1`] = `
<div>
  <br />
  <div
    className="container"
  >
    <h3
      className="text-center allRecipes-title"
    >
      An Error has ocurred. Please try again later
    </h3>
    <br />
  </div>
  <br />
  <Footer />
</div>
`;

exports[`ErrorBoundary component should render correctly 1`] = `undefined`;
